sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,115 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def detect_stack_drift(self):
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        :rtype: Tuple[str, Union[str, dict]]
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift(detection_id)
+
+        if status == "DETECTION_COMPLETE":
+            response = self._describe_stack_resource_drifts()
+            return_value = (self.stack.external_name, response)
+        else:
+            return_value = (self.stack.external_name, status)
+
+        return return_value
+
+    def _wait_for_drift(self, detection_id):
+        """
+        Waits for drift detection to complete.
+
+        :param detection_id: The drift detection ID.
+        :type detection_id: str
+
+        :returns: The drift status.
+        :rtype: str
+        """
+        timeout = 300
```

Comment:
I'm wondering what's the motivation behind this timeout. How confident are you that drift detection would not reasonably exceed 5 minutes?

Reply:
@jfalkenstein In my testing, I never saw this take more than 10-20 seconds or so. If the 5 minutes timeout is exceeded, then I assume something must really have gone wrong. The motivation is that I am waiting for something, and there's always a scenario where the timeout can be exceeded (if something is broken on AWS's side).

Reply:
Ok. I think 300 is an acceptable initial timeout. We'll get issues to fix it if we need to vary it.